### DISCUSSION TITLE
StrawberryShake - graphqlrc.json purpose on actual server

### DISCUSSION THREAD

**Discussion by mastery-sanket-nagare** [ChilliCream/hotchocolate#5645] [Q&A] [answered]
With StrawberryShake, what purpose does graphqlrc.json serve apart from the first-time configuration, and what does it perform on the actual live environment?

The reason I am asking this is, we are using StrawberryShake as a GraphQL client in our dev environment and have been successfully using it with all the tests passed. 
In graphqlrc.json configuration, we have the GQL URI of our dev environment and 
in Startup registration, we are setting the base URL dynamically while registering the AddHttpClient.

What we are trying to understand is, how the graphqlrc.json would work in other environments, such as UAT and Prod since we have the dev URI in there.

**Comment by tobias-tengler** [comment, marked as answer]
To my knowledge there are two main purposes:
1. The CLI provides the ability to update the GraphQL schema. This will use the endpoint specified in the file to re-download (update) the `schema.graphql` file.
2. It provides the path to the schema file to the GraphQL language server, so you get autocompletion and so on in your IDE.
3. It lets you configure how the graphql client is generated over `extensions`

It has no influence at runtime.
